src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java

```diff
@@ -738,6 +741,21 @@ static String getPropertyOnlyName(String name, Logger logger) {
         return result;
     }
 
+    static synchronized void addConnRef(UUID clientConnectionId) {
+        if (null != connRef && !connRef.contains(clientConnectionId)) {
```

Comment:
Adding an existing element to a hashset is a "no-op". We don't need to check whether the elements exists or not, in fact, we save performance by not doing so.